src/PointwiseFunctions/AnalyticData/GrMhd/MagnetizedTovStar.hpp

```diff
@@ -194,32 +143,68 @@ class MagnetizedTovStar : public virtual evolution::initial_data::InitialData,
   using tov_star = RelativisticEuler::Solutions::TovStar;
 
  public:
-  struct PressureExponent {
-    using type = size_t;
+  struct PoloidalField {
     static constexpr Options::String help = {
-        "The exponent n_s controlling the smoothness of the field"};
+        "Options related to the poloidal component of initial magnetic field."};
   };
 
-  struct VectorPotentialAmplitude {
+  struct PoloidalPressureExponent
```

Comment:
Is it not possible to have a tag that's a `std::vector<std::unique_ptr<InitialMagneticField>>` and then just superpose them? That seems like the most flexible implementation (and also the least code)